benchmark/src/DriverBench/ParallelBench.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+namespace MongoDB\Benchmark\DriverBench;
+
+use Amp\Parallel\Worker\DefaultPool;
+use Generator;
+use MongoDB\Benchmark\Fixtures\Data;
+use MongoDB\Benchmark\Utils;
+use MongoDB\BSON\Document;
+use MongoDB\Collection;
+use PhpBench\Attributes\AfterClassMethods;
+use PhpBench\Attributes\BeforeClassMethods;
+use PhpBench\Attributes\BeforeMethods;
+use PhpBench\Attributes\ParamProviders;
+use PhpBench\Attributes\Revs;
+use RuntimeException;
+
+use function Amp\ParallelFunctions\parallelMap;
+use function Amp\Promise\wait;
+use function array_map;
+use function ceil;
+use function count;
+use function file;
+use function file_get_contents;
+use function file_put_contents;
+use function is_dir;
+use function mkdir;
+use function pcntl_fork;
+use function pcntl_waitpid;
+use function range;
+use function sprintf;
+use function str_repeat;
+use function sys_get_temp_dir;
+use function unlink;
+
+use const FILE_IGNORE_NEW_LINES;
+use const FILE_NO_DEFAULT_CONTEXT;
+use const FILE_SKIP_EMPTY_LINES;
+
+/**
+ * For accurate results, run benchmarks on a standalone server.
+ *
+ * @see https://github.com/mongodb/specifications/blob/ddfc8b583d49aaf8c4c19fa01255afb66b36b92e/source/benchmarking/benchmarking.rst#parallel
+ */
+#[BeforeClassMethods('beforeClass')]
+#[AfterClassMethods('afterClass')]
+final class ParallelBench
+{
+    /** @var string[] */
+    private static array $files = [];
+
+    public static function beforeClass(): void
+    {
+        // Generate files
+        $fileContents = str_repeat(file_get_contents(Data::LDJSON_FILE_PATH), 5_000);
+        foreach (self::getFileNames() as $file) {
+            file_put_contents($file, $fileContents);
+        }
```

Comment:
I like how you prepare the files for each run instead of relying on the archive that contains a few hundred MB of the same data 👍